notebook/converted/batch_effect_correction.py

```diff
@@ -0,0 +1,424 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Evaluation of Batch Effect Correction
+# This notebook asks whether batch effect correction works for correcting inter-study differences and/or interplatform differences.
+
+# In[1]:
+
+
+import collections
+import json
+import os
+import pickle
+from typing import Text, Dict, Union, List
+
+import pandas as pd
+import sklearn
+from plotnine import ggplot, geom_point, aes
+from tqdm.notebook import tqdm
+
+from saged import utils
+
+
+# In[2]:
+
+
+def return_unlabeled():
+    # For use in a defaultdict
+    return 'unlabeled'
+
+
+# In[3]:
+
+
+data_dir = '../../data/'
+map_file = os.path.join(data_dir, 'sample_classifications.pkl')
+
+sample_to_label = utils.parse_map_file(map_file)
+sample_to_label = collections.defaultdict(return_unlabeled, sample_to_label)
+
+with open(map_file, 'rb') as in_file:
+    label_to_sample = pickle.load(in_file)[0]
+
+
+# In[4]:
+
+
+metadata_path = os.path.join(data_dir, 'aggregated_metadata.json')
+metadata = None
+with open(metadata_path) as json_file:
+    metadata = json.load(json_file)
+
+
+# ## Get blood samples from dataset
+# 
+
+# In[5]:
+
+
+sample_ids = utils.get_blood_sample_ids(metadata, sample_to_label)
+print(len(sample_ids))
+
+
+# In[6]:
+
+
+compendium_path = os.path.join(data_dir, 'HOMO_SAPIENS.tsv')                                                             
+
+# Not all labeled samples show up in the compendium, which causes pandas to panic. To fix this  
+# we have to take the intersection of the accessions in sample_ids and the accessions in the       
+# compendium                                                                                    
+header_ids = None                                                                               
+with open(compendium_path) as in_file:                                                          
+    header = in_file.readline()                                                                 
+    header_ids = header.split('\t')                                                             
+
+valid_sample_ids = [id_ for id_ in sample_ids if id_ in header_ids]  
+valid_sample_ids = set(valid_sample_ids)
+print(len(valid_sample_ids))
+
+
+# ## Read in expression data
+# Now that we know which samples we're interested in, let's read those samples from the compendium.
+# This is doable with pandas, but subsetting a dataframe is extremely slow for some reason.
+# Manually writing the subsetted tsv to a file and loading it should be substantially faster and
+# less memory intensive
```

Comment:
The human compendium is over 100GB on disk, and would take up a little more space in memory. As a result it makes sense to write the subsetted rows directly to a file rather than getting pandas involved